lib/src/facts/posix/uptime_resolver.cc

```diff
@@ -43,6 +48,22 @@ namespace facter { namespace facts { namespace posix {
 
     int64_t uptime_resolver::get_uptime()
     {
+        // We first try to get the system boot time by reading it directly from
+        // the utmp file using the getutxid system call.
+        LOG_DEBUG("Attempting to calculate the uptime from the utmp file...");
+        struct utmpx query;
+        query.ut_type = BOOT_TIME;
+        setutxent();
+        struct utmpx* ent = getutxid(&query);
+        if (ent) {
+          int64_t boot_time = ent->ut_tv.tv_sec;
+          endutxent();
```

Comment:
I'd kind of like to see a nice RAII wrapper for `setutxent`/`endutxent`, in order to avoid having to manually deal with this cleanup in multiple code paths. Even something as simple as: 
```
struct utmpx {
    utmpx() { setutxent(); }
    ~utmpx() { endutxent(); }
    utmpx* utxid(utmpx* query) { return getutxid(query); }
}
```
We could also then add error checking to ensure only one instance exists at once (since this API deals with global state)